src/frontend/src/pages/CreateChangeRequestPage/CreateChangeRequest.tsx

```diff
@@ -44,31 +44,46 @@ const CreateChangeRequest: React.FC<CreateChangeRequestProps> = () => {
 
   const { userId } = auth.user;
 
+  const addProposedSolutions = async (crId: number) => {
+    proposedSolutions.forEach(async (ps) => {
+      const { description, timelineImpact, scopeImpact, budgetImpact } = ps;
+
+      await cpsMutateAsync({
+        crId,
+        submitterId: userId,
+        description,
+        timelineImpact,
+        scopeImpact,
+        budgetImpact
+      });
+    });
+  };
+
   const handleConfirm = async (data: FormInput) => {
+    let crId: number;
     try {
       const cr = await mutateAsync({
         ...data,
         wbsNum: validateWBS(wbsNum)
       });
-      const crId = parseInt(cr.message);
-      proposedSolutions.forEach(async (ps) => {
-        const { description, timelineImpact, scopeImpact, budgetImpact } = ps;
-
-        await cpsMutateAsync({
-          crId,
-          submitterId: userId,
-          description,
-          timelineImpact,
-          scopeImpact,
-          budgetImpact
-        });
-      });
-
-      history.push(routes.CHANGE_REQUESTS);
+      crId = parseInt(cr.message);
     } catch (e) {
       if (e instanceof Error) {
+        if (e.message.startsWith('Failed to send slack notification for CR: ')) {
+          const splitMessage = e.message.split(': ');
+          crId = parseInt(splitMessage[1]);
```

Comment:
this is too janky, I see why you did it, but I'm not gonna add this code to the frontend. Ideally how this should be handled is that all the info should just go to one endpoint that creates the CR, then adds the proposed solutions, then sends the slack ping. I don't even think it'd be that hard to do cause you can probably just call the proposed solution service method from the CR service method